yarn-project/aztec.js/src/account_impl/single_key_account_contract.ts

```diff
@@ -48,7 +48,7 @@ export class SingleKeyAccountContract implements AccountImplementation {
     const txRequest = TxExecutionRequest.from({
       argsHash: packedArgs.hash,
       origin: this.address,
-      functionData: new FunctionData(selector, true, false),
+      functionData: new FunctionData(selector, abi.isInternal, true, false),
```

Comment:
perhaps there should be a helper function to generate the function data from the ABI, this code appears alot and could be derived. Above the selector is calculated from two other fields on the ABI, my feeling is that all of it could and should be calculated from there

This is probably a bit too much to squeeze into this pr, but i think it should be done if you agree

Reply:
Ye, think that would be nice to have in a separate pr.

Reply:
Created issue https://github.com/AztecProtocol/aztec-packages/issues/1223